### PR TITLE
Add has backup flowcell

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Addresses #2507. Add a new boolean, non-nullable field `has_backup` to the `Flowcell` store model with `False` as default.

### Added

- Field `has_backup` to the `Flowcell` store model
- optional parameter `has_backup` to all the Flowcell-creating functions.
- Migration revision file for the changes in the database tables.

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] `alembic -c path/to/alembic.ini upgrade head`
- [x] `alembic -c path/to/alembic.ini downgrade -1`

### Expected test outcome

- [x] Check that the column `has_backup` appears in the Flowcell table with 0 for NovaSeqX and 1 for the rest
- [x] Check that the column disappears after the downgrade

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO, HS
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MINOR** - when you add functionality in a backwards compatible manner

## Implementation Plan

- [ ] Deploy this branch on 
  - [ ] cg stage
  - [ ] cg production

### CODE REVIEW

File: alembic/versions/c3da223e60d8_add_has_backup_field.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    # ### commands auto generated by Alembic - please adjust! ###
+    bind = op.get_bind()
+    session = Session(bind=bind)
+
+    op.alter_column(
+        table_name="application",
+        column_name="min_sequencing_depth",
+        existing_type=mysql.INTEGER(),
+        nullable=False,
+    )
```
Comment (diitaz93) [inline alembic/versions/c3da223e60d8_add_has_backup_field.py:34]: This automatically generated alteration was unexpected. I see that in #1904 this field was modified to be non-nullable but the database doesn't seem to have been modified.

**Review by ChrOertlin — APPROVED** [review]
The flow cell part looks good, Not sure about the application thing.

File: alembic/versions/c3da223e60d8_add_has_backup_field.py (hunk 1 above)
Comment (ChrOertlin) [reply alembic/versions/c3da223e60d8_add_has_backup_field.py:34]: Does this need to be modified in the SQL schema?

Comment (diitaz93) [reply alembic/versions/c3da223e60d8_add_has_backup_field.py:34]: I am considering removing it. It is out of the scope of this PR but is an inconsistency that has to be addressed at some point.

**Review by henrikstranneheim — APPROVED** [review]
Looks good 👍

**Comment by diitaz93** [comment]
## Test on stage
Did the upgrade and the column appeared with the expected values:
Screenshot 2023-09-22 at 16 51 03

File: alembic/versions/c3da223e60d8_add_has_backup_field.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Session = sessionmaker()
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    bind = op.get_bind()
+    session = Session(bind=bind)
+
+    op.add_column(
+        table_name="flowcell", column=sa.Column("has_backup", sa.Boolean(), nullable=False)
```
Comment (Vince-janv) [inline alembic/versions/c3da223e60d8_add_has_backup_field.py:30]: Does this work? If we add a new column all the current entries will have value null right, which is in violation with `nullable=False`. If this part is tested please ignore the comment

**Review by Vince-janv — APPROVED** [review]
Nice 👍

File: alembic/versions/c3da223e60d8_add_has_backup_field.py (hunk 2 above)
Comment (diitaz93) [reply alembic/versions/c3da223e60d8_add_has_backup_field.py:30]: It was tested on stage and went well 🙂 The field has a default of `False` and the code just below updates the values.

**Comment by diitaz93** [comment]
## Deployed to stage:
```shell
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/js.diazboada
done.
Log deploy... done.
cg, version 49.5.0
θ61° [js.diazboada@hasta:~] [S_base] 2m54s
```
## Deployed to production:
```shell
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/js.diazboada
done.
Log deploy... done.
cg, version 49.5.0
[js.diazboada@hasta:~] [P_base] 2m14s
```

**Pull request merged**